apps/angular/permissions/src/app/routes.ts

```diff
@@ -1,14 +1,78 @@
-export const APP_ROUTES = [
+import { Route } from '@angular/router';
+import { Role } from './user.model';
+import { HasPermission } from './common/guards/is-admin.guard';
+export interface TypedRoute extends Route {
+  data?: {
+    isAdmin?: boolean;
+    roles?: Role[];
+  };
+}
+export const APP_ROUTES: TypedRoute[] = [
   {
     path: '',
     loadComponent: () =>
       import('./login.component').then((m) => m.LoginComponent),
   },
   {
     path: 'enter',
+    canMatch: [HasPermission],
```

Comment:
HasPermission is a function which can take parameters. Don't use the data property

Reply:
@tomalaforge  u mean i can call that function directly as well ?

Reply:
That's the goal of the new API. Guard are function, not class,

Reply:
i will update that.